server/controllers/auth/signup.js

```diff
@@ -1,14 +1,34 @@
-// modules 
+// modules
+const Joi = require('@hapi/joi');
 const bcrypt = require('bcryptjs');
 const uuid = require('uuid/v1');
 const jwt = require('jsonwebtoken');
 
 const database = require("../../database");
 
+const signupSchema = Joi.object({
+	email: Joi.string()
+		.email()
+    .min(4)
+    .max(50)
+    .required(),
+	password: Joi.string()
+		.min(6)
+    .max(72)
+    .required(),
+});
+
 exports.signup = (req, res, next) => {
-  const email = req.body.email;
-  // password cannot be longer than 72 characters
-  const password = req.body.password;
+
+	const valid = signupSchema.validate(req.body);
+
+	if (valid.error) {
+		res.status(422);
```

Comment:
You are just sending the status code but not sending any response to the client?

Reply:
The way it works is, the `res` object at this line has status code. The next line has the `next()` method with error. Since there is no other middleware after this one, the code execution continues and reaches error handler middleware, and that's where the response is handled with this error code.

Reply:
I haven't seen that method to pass error to middleware further.

Reply:
`app.use()` is intended to use in specific order. Error handlers are placed at the very end for this reason. When no other middlewares respond, the error middleware takes over the response.